models/detr.py

```diff
@@ -139,7 +139,7 @@ def loss_boxes(self, outputs, targets, indices, num_boxes):
         assert 'pred_boxes' in outputs
         idx = self._get_src_permutation_idx(indices)
         src_boxes = outputs['pred_boxes'][idx]
-        target_boxes = torch.cat([t['boxes'][i] for t, (_, i) in zip(targets, indices)], dim=0)
+        target_boxes = torch.cat([t['boxes'][i.long()] for t, (_, i) in zip(targets, indices)], dim=0)
```

Comment:
I believe the boxes should stay as `float` otherwise the L1 loss and GIoU loss will potentially error

Reply:
Oh, looks like I was wrong, you are casting the indices, not the boxes, my bad.